### PR TITLE
Implement Refined Gen instances (#4602)

### PR DESCRIPTION
**Pull request by kazchimo**

### CODE REVIEW

File: test/shared/src/main/scala/zio/test/Gen.scala [hunk 1]
```diff
@@ -86,7 +86,7 @@ final case class Gen[-R, +A](sample: ZStream[R, Nothing, Sample[R, A]]) { self =
    * meet the specified predicate.
    */
   def filterNot(f: A => Boolean): Gen[R, A] =
-    filter(a => !(f(a)))
+    filter(a => !f(a))
```
Comment (adamgfraser) [inline test/shared/src/main/scala/zio/test/Gen.scala:89]: Stray change?

File: test/shared/src/main/scala/zio/test/Gen.scala [hunk 2]
```diff
@@ -478,7 +501,7 @@ object Gen extends GenZIO with FunctionVariants with TimeVariants {
    */
   def fromIterable[R, A](
     as: Iterable[A],
-    shrinker: (A => ZStream[R, Nothing, A]) = defaultShrinker
+    shrinker: A => ZStream[R, Nothing, A] = defaultShrinker
```
Comment (adamgfraser) [inline test/shared/src/main/scala/zio/test/Gen.scala:504]: Stray change?

File: test/shared/src/main/scala/zio/test/Gen.scala [hunk 3]
```diff
@@ -375,7 +398,7 @@ object Gen extends GenZIO with FunctionVariants with TimeVariants {
 
   /**
    * Combines the specified deterministic generators to return a new
-   * deterministic generator that generatoes all of the values generated by
+   * deterministic generator that generators all of the values generated by
```
Comment (adamgfraser) [inline test/shared/src/main/scala/zio/test/Gen.scala:401]:
```suggestion
   * deterministic generator that generates all of the values generated by
```

**Review by adamgfraser — CHANGES_REQUESTED** [review]
Looks great! Just a couple of minor comments.

File: test/shared/src/main/scala/zio/test/Gen.scala (hunk 1 above)
Comment (kazchimo) [reply test/shared/src/main/scala/zio/test/Gen.scala:89]: Minor refactoring.
Shouldn't I commit this change in this PR?

File: test/shared/src/main/scala/zio/test/Gen.scala (hunk 2 above)
Comment (kazchimo) [reply test/shared/src/main/scala/zio/test/Gen.scala:504]: Minor refactoring.
Shouldn't I commit this change in this PR?

File: test/shared/src/main/scala/zio/test/Gen.scala (hunk 1 above)
Comment (adamgfraser) [reply test/shared/src/main/scala/zio/test/Gen.scala:89]: No need to add additional parentheses to code that is already compiling.

**Review by adamgfraser — APPROVED** [review]

File: test/shared/src/main/scala/zio/test/Gen.scala (hunk 1 above)
Comment (kazchimo) [reply test/shared/src/main/scala/zio/test/Gen.scala:89]: Yeah, I removed unnecessary parentheses.

Comment (adamgfraser) [reply test/shared/src/main/scala/zio/test/Gen.scala:89]: 👍

**Pull request merged**